blog/_posts/2021-12-20-discord.md

```diff
@@ -0,0 +1,183 @@
+Discord server had become much more active than our Gitter rooms.
+
+Slack was considered as an option, but we decided it is best suited
+for private companies and groups, but not so well suited for
+open-ended public communities like ours.
+
+## Tell us more history.
+
+Okay, let's take a trip down memory lane.
+
+During Scala's earliest days (at least since 2007, perhaps earlier),
+the main chat platform for Scala was IRC.  In those early years, when
+the community was still small and there weren't that many Scala
+conferences and meetups yet, the IRC channel (along with the Google
+Groups mailing lists) was an important place where community members
+got to know each other and, one might even say, became a community.
+
+Later, the IRC channel declined in popularity.  In my personal view,
+this was due to a lack of moderation and lack of any code of conduct.
+It became essential to have these once the Scala community had become
+larger.
+
+The IRC channel still exists to this day, now on
+[Libera](https://libera.chat) instead of Freenode, but traffic there
+has slowed to a trickle in recent years.
+
+Gitter launched in 2014 and was widely adopted by the Scala community
+in 2015. The major advantages of Gitter over IRC include:
+
+* Scrollback: you can log out, log back on, and still see what
+  people were saying while you were away.  (On IRC, this was only
+  possible if you used special software.)
+* Markdown support, including syntax-highlighting of Scala code.
+* Messages can be edited (within a limited time frame) and
+  deleted.
+* More features supporting effective content moderation.
+
+Discord shares these advantages as well.
+
+Some advantages of Gitter that Discord _doesn't_ share:
+
+* GitHub integration, including GitHub-based login. (No need to
+  make a separate account.)
+* Easy, decentralized, room creation, with room names based on GitHub
+  repo names, making it a no-brainer for an open source maintainer to
+  create a room specifically for their library.
+
+However, in recent years, the Gitter platform largely stopped
+advancing and eventually came to feel stale.  After Gitter was
+acquired by GitLab in 2017, some features were added such as
+threading, but the implementation of threading was unpopular with
+users. The quality and feature set of the mobile client fell behind
+those of the desktop client.  New Vector Limited acquired Gitter in
+2020 and added Matrix support, but otherwise has not advanced the
+platform.
+
+Meanwhile, Discord was becoming extremely popular in the video gaming
+community. Some open-source communities were initially hesitant to
+adopt it because of the strong association with gaming (some companies
+block it for that reason). Discord also got some bad publicity, years
+ago, when some unsavory political groups used it for organizing, but
+the association of Discord with that episode in the public's
+perception has now largely faded.
```

Comment:
I wasn't even aware of that. Perhaps I'm not the only one. Is it worth mentioning this? It might re-create the association, even if faintly, for no good reason?

Reply:
That's a good question! I'm not sure what I think.

fwiw, it is buried pretty deeply in the TL;DR part of the post.

I'm glad to hear that you weren't even aware of it. Two or three years ago, it was definitely something that some people brought up in private discussions, but it hasn't come up again recently as far as I can recall.

I looked through https://users.scala-lang.org/t/can-we-have-an-official-scala-discord-server/5157 just now and I see that literally nobody mentioned this issue, so that would argue for not mentioning it here, either.

Reply:
+1 to @sjrd. Didn't know about it, and I'd vote to remove mentioning unsavory political anything in this post if we don't want to get this post itself involved in some political "how dare you"s.

Reply:
Okay, I've pushed a commit that takes it out. I also took out the "some companies
block it" bit since it's been a while since I've heard anyone mention or complain about that, either.